hedera-node/src/main/java/com/hedera/services/stream/RecordStreamFileWriter.java

```diff
@@ -489,6 +541,52 @@ private SignatureObject generateSignatureObject(final byte[] hash) {
 				.build();
 	}
 
+	private void createSidecarFile(
+			final SidecarFile.Builder sidecarFileBuilder,
+			final Instant firstTxnInstant,
+			final RecordStreamType.SidecarType sidecarType
+	) throws IOException, NoSuchAlgorithmException {
+		// create a file only if there are any sidecar records of this type for the current period
+		if (sidecarFileBuilder.getSidecarRecordsCount() > 0) {
+			final var sidecarFile = new File(generateSidecarFilePath(firstTxnInstant, sidecarType.getSidecarId()));
+			try (FileOutputStream stream = new FileOutputStream(sidecarFile, false);
+				 SerializableDataOutputStream dos = new SerializableDataOutputStream(new BufferedOutputStream(stream))
+			) {
+				// write contents of sidecar
+				dos.write(serialize(sidecarFileBuilder));
+
+				// make sure the whole sidecar is written to disk before continuing
+				// with calculating its hash and saving it as part of the SidecarMetadata
+				dos.flush();
+				stream.flush();
+				stream.getChannel().force(true);
+				stream.getFD().sync();
+
+				LOG.debug(OBJECT_STREAM_FILE.getMarker(),
+						"Sidecar sidecarFilePath written successfully {}", sidecarFile.getName());
+				recordStreamFileBuilder.addSidecars(createSidecarMetadata(sidecarFile, sidecarType));
+			}
+		}
+	}
+
+	private SidecarMetadata.Builder createSidecarMetadata(
+			final File sidecarFile,
+			final RecordStreamType.SidecarType sidecarType
+	) throws IOException, NoSuchAlgorithmException, IllegalStateException {
+		return SidecarMetadata.newBuilder()
+				.setHash(toProto(LinkedObjectStreamUtilities.computeEntireHash(sidecarFile).getValue()))
+				.setId(sidecarType.getSidecarId())
```

Comment:
This id should be a regular number that starts at one and and increments per sidecar within the record file. It's not hardcoded to an enum value. The only reason this increases and causes additional side sidecar files to be created is, in my opinion:

* If the number of items in the files exceeds some configurable threshold
* Operators ask for separation of certain types since they don't want to process all

Reply:
Fixed [here](https://github.com/hashgraph/hedera-services/pull/3625/commits/578e6eeff92dcb28fd743d055a6320db034b6181), hardcoded id to 1, instead of keeping track with a variable since we will be creating only a single sidecar file for now.